### ISSUE TITLE
Support "x-ms-format": "duration-constant"

### ISSUE THREAD

**Issue by pshao25** [Azure/typespec-azure#1901]
Some swagger will have `"x-ms-format": "duration-constant"` on their model property for example. We may need a new scalar for it. What's more, TCGC needs to pass this value to emitter. 

BTW I have some trials just FYI
1. 
```
model Test {
  @format("duration-constant")
  input: duration;
}
```
This gives error `Cannot apply @format decorator to type it is not a string`

2. 
```
model Test {
  @format("duration-constant")
  input: string;
}
```
This gives warning `'string' format 'duration-constant' is not supported in Autorest. It will not be emitted`. The emitted swagger does not emit this either.

3.
```
model Test {
  @encode("duration-constant")
  input: duration;
}
```
This emits
```
"input": {
  "type": "string",
  "format": "duration-constant"
}
```
SDK has this error:
```
fatal   | System.InvalidOperationException: The MaximumDuration property of VirtualMachineInstallPatchesParameters ends with "Duration" but does not use the duration format to be generated
   as TimeSpan type. Add "format": "duration" with directive in autorest.md for the property if it's ISO 8601 format like P1DT2H59M59S. Add "x-ms-format": "duration-constant" if it's the con 
  stant format like 1.2:59:59.5000000. If the property does not conform to a TimeSpan format, please use "x-ms-client-name" to rename the property for the client.
```
In other words, `"x-ms-format": "duration-constant"` is a must in the generated swagger.

**Comment by pshao25** [comment]
Summary for Option3:

- Use encode to express duration-constant in TypeSpec
- C# can check this string and do whatever is needed
- Other emitters must not crash, and either ignore (and return string) or decide to handle it like C#
- We need to document this somewhere in a list of acceptable encoding
- Swagger compat should be done with directives in autorest.md

**Comment by weidongxu-microsoft** [comment]
For
```
model Test {
  @encode("duration-constant")
  input: duration;
}
```

TCGC gives
```
                                  type:
                                    kind: duration
                                    decorators: []
                                    name: duration
                                    encode: duration-constant
                                    wireType:
                                      kind: string
                                      decorators: []
                                      name: string
                                      doc: A sequence of textual characters.
                                      crossLanguageDefinitionId: TypeSpec.string
                                    doc: A duration/time period. e.g 5s, 10h
                                    crossLanguageDefinitionId: TypeSpec.duration
```

[playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvdmVyc2lvbmluZyI7CskfYXp1cmUtdG9vbHMvyCstxhVjb3JlIjsKCnVzaW5nIFR5cGVTcGVjLlJlc3Q70BVIdHRw0RVWyXLIG0HEWS5Db3JlLlRyYWl0czsKCkBzZXJ2aWNlKHsKICB0aXRsZTogIlRleHQgYXV0aG9y5ACxLAp9KQpA5wDCZWQoU8Yzx2FzKQpuYW1lc3BhY2UgxDpByDk7CgplbnVtIM8wIMRrQHVzZURlcGVuZGVuY3ko6wCbyCcudjFfMF9QcmV2aWV3XzIpCiAgdjE6ICIyMDIxLTDEA%2BQAmQoKaW50ZXJmxH1SZXNvdXJjZU9wZXJhdMRECiAgZXh0ZW5kc%2BwA%2FdIoPE5vUmVwZWF0YWJsZVJlcXVlc3RzICYKIMUBTm9Db25kacQqYWzUHmxpZW50xxlJZD4ge30KCkByx2YoInByb2plY3RzIikKbW9kZWwgUMYR5gEJa2V5xAd2aXNpYmlsaXR5KCJyZWFkIuQA8WlkOiBzdOcBTCAgZGVzY3JpcOQAiT%2FKGSAg5AGAyxHET2VuY29kZSgiZHXmAOMtY29uc3RhbnTGWG5wdXQ%2FOiDIHjvuAUnpAcLlAKJsaXN0IGlz8wFf6QFKTGlzdDznANY%2BxE0%3D&e=%40azure-tools%2Ftypespec-client-generator-core&options=%7B%7D)

Without feature support, I guess SDK would have client type Duration (not String).

**Comment by abatishchev** [comment]
hi @pshao25 can you please revisit https://github.com/microsoft/typespec/issues/3318?
